lib/kafo/kafo_configure.rb

```diff
@@ -143,6 +143,8 @@ def initialize(*args)
       # so we limit parsing only to app config options (because of --help and later defined params)
       parse clamp_app_arguments
       parse_app_arguments # set values from ARGS to config.app
+      config.configure_application
```

Comment:
Why is it needed to save the configuration at this point?

Reply:
If a user passes in arguments from the command line, we need to ensure that those values, for any options that are marked for persistence in the config file, get saved to the config file.

Reply:
It feels wrong to save the config before running the checks etc. I tried to look in the docs but it doesn't say anything about it. Perhaps it should only be saved after `pre_commit`?

My reasoning is that the answers aren't stored until then either and it's still possible to abort the run without making modifications.

Reply:
I think that's fair